tests/lisa/test_wlgen.py

```diff
@@ -0,0 +1,379 @@
+            },
+            # Ramp phases:
+            {
+                "loop": 20,
+                "run": 5000,
+                "timer": {
+                    "period": 50000,
+                    "ref": "task_ramp"
+                }
+            },
+            {
+                "loop": 20,
+                "run": 15000,
+                "timer": {
+                    "period": 50000,
+                    "ref": "task_ramp"
+                }
+            },
+            {
+                "loop": 20,
+                "run": 25000,
+                "timer": {
+                    "period": 50000,
+                    "ref": "task_ramp"
+                }
+            },
+            {
+                "loop": 20,
+                "run": 35000,
+                "timer": {
+                    "period": 50000,
+                    "ref": "task_ramp"
+                }
+            },
+            {
+                "loop": 20,
+                "run": 45000,
+                "timer": {
+                    "period": 50000,
+                    "ref": "task_ramp"
+                }
+            },
+            # Heavy phase:
+            {
+                "loop": 1,
+                "run": 90000,
+                "timer": {
+                    "period": 100000,
+                    "ref": "task_ramp"
+                }
+            }]
+
+        self.assertListEqual(phases, exp_phases)
+
+        rtapp.run(out_dir=self.host_out_dir)
+
+        rtapp_cmds = [c for c in self.target.executed_commands if 'rt-app' in c]
+        self.assertListEqual(rtapp_cmds, [self.get_expected_command(rtapp)])
+
+        self.assert_output_file_exists('output.log')
+        self.assert_output_file_exists('rt-app-task_ramp-0.log')
+        self.assert_output_file_exists('test_00.json')
+
+
+class TestRTACustom(RTABase):
+    def test_custom_smoke(self):
+        """
+        Test RTA custom workload
+
+        Creates an rt-app workload using 'custom' and checks that the json
+        roughly matches the file we provided. If we have root, attempts to run
+        the workload.
+        """
+
+        json_path = os.path.join(os.getenv('LISA_HOME'),
+                                 'assets', 'mp3-short.json')
+        rtapp = RTA(self.target, name='test', calibration=self.calibration)
+
+        # Configure this RTApp instance to:
+        rtapp.conf(kind='custom', params=json_path, duration=5,
+                   run_dir=self.target_run_dir)
+
+        with open(rtapp.json) as f:
+            conf = json.load(f)
+
+        # Convert to str because unicode
+        tasks = set([str(k) for k in conf['tasks'].keys()])
+        self.assertSetEqual(
+            tasks,
+            set(['AudioTick', 'AudioOut', 'AudioTrack',
+                 'mp3.decoder', 'OMXCall']))
+
+        # Would like to try running the workload but mp3-short.json has nonzero
+        # 'priority' fields, and we probably don't have permission for that
+        # unless we're root.
```

Comment:
Agreed. Priorities is one of these things that we just won't be able to test using a localhost target.